add: useContractFunctionWithDefaultFees

sirpy opened the pull request:
# Description
- added new hook that wraps useContractFunction send with fixed gas price
- replaced all useContractFunction usages

About /GoodProtocolUI/issues/335

L03TJ3 reviewed (changes requested):
Works everywhere
except microbridge where for some reason the gaslimit has no set value
(not fully understanding where it goes wrong)

Reproducible with the MicroBridge story in good-design.

sirpy commented:
@L03TJ3 i've fixed the bridge. i was missing a paramter when encoding the transferAndCall. Usually when you see a gas issue in metamask it is because the dapp failed to estimate the gas because the TX is going to revert.

I've also fixed the bridge to work with correct token decimals. I've started using CurrencyValue more and refactored to simplify the way you handled the tokens decimals. please review and that i'm not breaking anything.

i've also tried to improve references between good-design and sdk-v2 in typescript and tried to fix the types issue by adding custom.d.ts to typeroots in tsconfig

sirpy commented on packages/sdk-v2/src/sdk/constants.ts:
```diff
@@ -1,139 +1,162 @@
+  const key = tokenName + "_" + chainId + "_" + decimals;
+  CURRENCIES_CASH[key] =
+    CURRENCIES_CASH[key] ||
+    new Currency(name, ticker, decimals, {
```
reverted
